### PR TITLE
$choice command

### PR DESCRIPTION
**Pull request by Nhlest**
# Description

Adds $choice command that returns a randomly chosen word from the list of arguments given.

Who asked: me :)
Issue: N/A

### CODE REVIEW

**Comment by pajlada** [comment]
Looks good, will test and make a final review tomorrow 👌

**Comment by Nhlest** [comment]
┌[ ◔ ͜ ʖ ◔ ]┐

File: internal/commands/choice/choice.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+func New() *Command {
+	return &Command{
+		Command: basecommand.New(),
+	}
+}
+
+func (c *Command) Run(s *discordgo.Session, m *discordgo.MessageCreate, parts []string) pkg.CommandResult {
+	parts = parts[1:]
+	response := fmt.Sprintf("%s, %s", m.Author.Mention(), parts[rand.Intn(len(parts))])
```
Comment (pajlada) [inline internal/commands/choice/choice.go:33]: When figuring out which part has been chosen, you need to ensure the list isn't empty.

Make sure the command doesn't crash with 0 arguments (i.e. `$choice`) and make sure the command doesn't run when only 1 argument is specified (i.e. `$choice foo`).

File: internal/commands/choice/choice.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	s.ChannelMessageSend(m.ChannelID, response)
```
Comment (pajlada) [inline internal/commands/choice/choice.go:34]: Make sure we trim out mentions here. You should not be able to do `$choice @everyone @everyone` or `$choice @pajlada @pajlada`

**Review by pajlada — CHANGES_REQUESTED** [review]

**Comment by pajlada** [comment]
@Nhlest I think it's best to use https://pkg.go.dev/github.com/bwmarrin/discordgo#Session.ChannelMessageSendComplex to remove mentions instead, you'll need to make sure the `AllowedMentions` parameter https://pkg.go.dev/github.com/bwmarrin/discordgo#MessageAllowedMentions doesn't allow any mentions. Once you've switched to it, I can try it out

**Comment by Nhlest** [comment]
@pajlada this way if user will try to use $choice word word2 word3 @mention word4 command will misfire 1/5 of the time. Seems like a strange behavior to have in my opinion.

**Review by pajlada — APPROVED** [review]

**Comment by pajlada** [comment]
Works for me
image

Known issues:
 - Commands currently give empty values as part of `parts`
so it could be `["a", "b", " ", " ", "c"]` - this is not a bug of this module, rather of the command matcher.
 - Disabling mentions also means that the person running the command misses the mention. This could be remediated later by using the reply functionality. Not necessary for this PR though

**Pull request merged**